gensim/models/word2vec.py

```diff
@@ -509,9 +510,16 @@ def build_vocab(self, sentences, keep_raw_vocab=False, trim_rule=None, progress_
         Each sentence must be a list of unicode strings.
 
         """
+        for sentence in sentences:
+            if hasattr(sentence, "words"):
+                if isinstance(sentence.words, string_types):
+                    warnings.warn("Sentences should be a list of unicode strings. True string provided.")
+                    break
         self.scan_vocab(sentences, progress_per=progress_per, trim_rule=trim_rule)  # initial survey
         self.scale_vocab(keep_raw_vocab=keep_raw_vocab, trim_rule=trim_rule)  # trim by min_count & precalculate downsampling
         self.finalize_vocab()  # build tables & arrays
+        self.check_vocab() # raise warning if vocab consists of all single characters
```

Comment:
So would it be better to just do away with this test? In any case the flow of scan->scale->finalize will have to be followed right? The check will already happen in scan then....